realm/src/main/java/io/realm/Realm.java

```diff
@@ -1276,6 +1276,7 @@ boolean hasChanged() {
     public void refresh() {
         checkIfValid();
         transaction.advanceRead();
+        sendNotifications();
```

Comment:
I see...
Is it possible to check the version then call `sendNotification`? I have seen @cmelchior made a change to return the version by `advanceRead()`